src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/LanguageServices/Handlers/CommandLineNotificationHandler.cs

```diff
@@ -29,9 +30,9 @@ public CommandLineNotificationHandler(UnconfiguredProject project)
 
         public void Handle(IWorkspaceProjectContext context, IComparable version, BuildOptions added, BuildOptions removed, ContextState state, IManagedProjectDiagnosticOutputService logger)
         {
-            foreach (Lazy<Action<string?, BuildOptions, BuildOptions>, IOrderPrecedenceMetadataView> value in CommandLineNotifications)
+            foreach (Action<string?, BuildOptions, BuildOptions> value in CommandLineNotifications.ExtensionValues())
             {
-                value.Value(context.BinOutputPath, added, removed);
+                value(context.BinOutputPath, added, removed);
```

Comment:
This question is for the sake of my own learning 
Why you need this change? is that to gain code cleanness or performance gain?

Reply:
`ExtensionValues` makes the code here a little cleaner, and it also ensures that any MEF exports that might throw when instantiated are filtered out, rather than causing this code to crash.

I guess when I said there was no behaviour change, I should have called this out.

Reply:
Here's the code in CPS (MS internal): https://dev.azure.com/devdiv/DevDiv/_git/CPS?path=/src/Microsoft.VisualStudio.ProjectSystem/Composition/Utilities/MefExtensions.cs&version=GBmain&line=286&lineEnd=327&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents